contracts/tge/HATAirdropFactory.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+// Disclaimer https://github.com/hats-finance/hats-contracts/blob/main/DISCLAIMER.md
+
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/proxy/Clones.sol";
+import "./HATAirdrop.sol";
+
+contract HATAirdropFactory is Ownable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    address public implementation;
+
+    event ImplementationUpdated(address indexed _newImplementation);
+    event TokensWithdrawn(address indexed _owner, uint256 _amount);
+    event HATAirdropCreated(
+        address indexed _hatAirdrop,
+        string _merkleTreeIPFSRef,
+        bytes32 _root,
+        uint256 _startTime,
+        uint256 _deadline,
+        uint256 _lockEndTime,
+        uint256 _periods,
+        uint256 _totalAmount,
+        IERC20Upgradeable _token,
+        ITokenLockFactory _tokenLockFactory
+    );
+
+    constructor (address _implementation) {
+        implementation = _implementation;
+    }
+
+    function updateImplementation(address _newImplementation) external onlyOwner {
+        implementation = _newImplementation;
+        emit ImplementationUpdated(_newImplementation);
+    }
+
+    function withdrawTokens(IERC20Upgradeable _token, uint256 _amount) external onlyOwner {
+        address owner = owner();
+        _token.safeTransfer(owner, _amount);
+        emit TokensWithdrawn(owner, _amount);
+    }
+
+    function createHATAirdrop(
+        string memory _merkleTreeIPFSRef,
+        bytes32 _root,
+        uint256 _startTime,
+        uint256 _deadline,
+        uint256 _lockEndTime,
+        uint256 _periods,
+        uint256 _totalAmount,
+        IERC20Upgradeable _token,
```

Comment:
do we want to airdrop tokens different than hats? IF not, perhaps this argument should be in the constructor?

Reply:
I'm not sure if we might do different tokens or not